src/components/profile/thread.js

```diff
@@ -14,13 +14,21 @@ type Props = {
     error: ?string,
   },
   setName: Function,
+  markAsDeleted: Function,
 };
 
 class ThreadWithData extends React.Component<Props> {
   componentWillMount() {
-    const { data: { thread }, setName } = this.props;
+    const { data: { thread }, data, setName, markAsDeleted } = this.props;
+
     if (setName && thread) {
-      this.props.setName(thread.community.name);
+      setName(thread.community.name);
+    }
+
+    // if the query finished loading but no thread was returned,
+    // clear this notification out
+    if (data.networkStatus === 7 && !data.thread && markAsDeleted) {
+      markAsDeleted();
```

Comment:
This will only ever fire on `mount`, where the network request might not have completed yet and thusly `data.networkStatus` might not be `7` yet. This should probably happen in `componentWillReceiveProps`?

Reply:
I think you can use the networkStatus enum exposed [here](https://github.com/apollographql/apollo-client/blob/b7fe71e9a0c2e32cea6553a9521ad66cf841f0cc/packages/apollo-client/src/core/networkStatus.ts) instead of the magic number. 

Although I’m not sure if thats even really exposed by Apollo and tbh I know next to nothing about typescript hehe

Reply:
So the code actually works due to the way it's composed upstream :) Feel free to pull this and try